### ISSUE TITLE
Reproducing the results in the GMN paper

### ISSUE THREAD

**Issue by zhengqigao** [deepmind/deepmind-research#30]
Hi, thanks for releasing the GMN code, which is really well written ! I am currently trying to use GMN, and when I run the sample code, I notice that the auc on validation is around 70%. Could u please give some hints about how to reproduce the results in the paper (i.e., auc around 90% for the GED task)? 

Would increasing training step works? If so, how large should it be? Thanks !

**Comment by yujiali** [comment]
Hi, if you train the model for 500k steps it should reach a reasonably good performance.  If you really care about this task you can train as much as you can afford and we haven't observed any overfitting as you can generate a enormous amount of training data.

**Comment by zhengqigao** [comment]
Yeah, there is no overfitting ! Would u mind to tell a little more about other hyper-parameters including but not limited to batch_size? Since I still cannot reach 90%.

**Comment by yujiali** [comment]
Hi, the batch size we used was 50 for this task, and the best performance was achieved sweeping over a few things, most importantly: training mode (pair vs triplet), share message passing layer parameters or not.  When training with pairwise loss, the margin was set to 0.1, and when training with triplet loss the margin was set to 1.0.  Learning rate etc. are just set to the default in the code.

I no longer remember the exact combination that led to the best performing model, but a grid search over these things should be able to get you close to the reported performance.

**Comment by yujiali** [comment]
I have just rerun the experiments and was able to reproduce the paper results or better.  The most important hyperparameter for this task / metric is training mode, you should set it to triplet.  Training on pairs is quite a bit worse than training on triplets up to 500k steps.

In our experiments we set the batch size to 50 but this does not play as significant of a role as the training mode.
